tasties_app/views.py

```diff
@@ -19,12 +19,31 @@ def base(request):
 
 def recipes(request):
     recipes_list = Recipe.objects.all()
+    categories_list = Category.objects.all()
+    selected_category = request.GET.get("category")
     recipes_with_ratings = {}
```

Comment:
We do not need this dictionary `recipes_with_ratings`. Instead we can query an annotated QuerySet, like this:
`recipes_list = Recipe.objects.annotate(recipe_rating=Avg('rating__rating')).order_by('-recipe_rating')` (of course, this will be the first definition of `recipes_list`)